### PR TITLE
Re-readd root/admin user detection

### PR DESCRIPTION
**Pull request by NoahvdAa**
This PR re-adds the root reverted in e8830b27. The check was throwing false positives because of a OpenJDk bug (no issue # yet, will add here later). This PR uses a different method for getting the UID (executing the id command), which isn't affected by the openjdk issue.

### CODE REVIEW

**Comment by clrxbl** [comment]
Why run `id -u` with the username specifically?
The command should already return the current user's ID.

**Comment by Prof-Bloodstone** [comment]
Wondering if it'd be better to use available API as before, and if it returns root as user, then use the command to confirm it?
This way it could potentially be faster and doesn't require for the `id` binary to be on the system most of the time.

File: patches/server/0814-Add-root-admin-user-detection.patch [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
++
++    static {
++        if (SystemUtils.IS_OS_WINDOWS) {
++            RUNNING_AS_ROOT_OR_ADMIN = Set.of(new NTSystem().getGroupIDs()).contains(WINDOWS_HIGH_INTEGRITY_LEVEL);
++        } else {
++            boolean isRunningAsRoot = false;
++            if (new UnixSystem().getUid() == 0) {
++                // Due to an OpenJDK bug, UnixSystem#getUid incorrectly returns 0 when the user doesn't have a username.
++                // Because of this, we'll have to double-check if the user ID is actually 0 by running the id command.
++                try (final InputStream inputStream = Runtime.getRuntime().exec("id -u").getInputStream()) {
```
Comment (Proximyst) [inline patches/server/0814-Add-root-admin-user-detection.patch]: Should use processes.

**Review by Proximyst — CHANGES_REQUESTED** [review]

File: patches/server/0814-Add-root-admin-user-detection.patch (hunk 1 above)
Comment (NoahvdAa) [reply patches/server/0814-Add-root-admin-user-detection.patch]: Is there an advantage to spawning a process with something like a ProcessBuilder (assuming that's what you mean here) vs spawning one via `Runtime#exec`?

**Comment by me4502** [comment]
How common is it for servers to not contain the id binary?

I'd question whether it's better to have it default to warning when it's missing - but guess it depends how often it's missing 

Logic being a host can easily fix it themselves if it shows up due to missing id binary

File: patches/server/0814-Add-root-admin-user-detection.patch [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
++    private static final boolean RUNNING_AS_ROOT_OR_ADMIN;
++    private static final String WINDOWS_HIGH_INTEGRITY_LEVEL = "S-1-16-12288";
++
++    static {
++        if (SystemUtils.IS_OS_WINDOWS) {
++            RUNNING_AS_ROOT_OR_ADMIN = Set.of(new NTSystem().getGroupIDs()).contains(WINDOWS_HIGH_INTEGRITY_LEVEL);
++        } else {
++            boolean isRunningAsRoot = false;
++            if (new UnixSystem().getUid() == 0) {
++                // Due to an OpenJDK bug, UnixSystem#getUid incorrectly returns 0 when the user doesn't have a username.
```
Comment (lynxplay) [inline patches/server/0814-Add-root-admin-user-detection.patch]: Should probably now also reference the [openjdk issue](https://bugs.openjdk.java.net/browse/JDK-8274721) to ease traceability/link to more information as to why this double check had to be implemented.

**Review by lynxplay — CHANGES_REQUESTED** [review]

**Review by me4502 — APPROVED** [review]

**Review by Proximyst — APPROVED** [review]

**Pull request merged**